Sources/NIOEmbedded/Embedded.swift

```diff
@@ -63,6 +67,7 @@ public final class EmbeddedEventLoop: EventLoop {
     /// The current "time" for this event loop. This is an amount in nanoseconds.
     /* private but tests */ internal var _now: NIODeadline = .uptimeNanoseconds(0)
 
+    private var scheduledTaskCounter = NIOAtomic.makeAtomic(value: UInt64(0))
```

Comment:
It's a shame that this now regresses a bunch of allocation counting tests that use `EmbeddedEventLoop` (and will no doubt regress allocation counting tests in other swift-nio-* packages). I wonder if we should make the counter for embedded loops global (or static) to reduce the noise of this one.

Reply:
Actually `EmbeddedEventLoop` isn't thread safe, can we just use a `UInt64`?

Reply:
We can, and probably should.